src/Microsoft.OData.Core/Uri/SelectExpandClauseToStringBuilder.cs

```diff
@@ -104,7 +107,63 @@ public override string Translate(WildcardSelectItem item)
         /// <returns>Defined by the implementer</returns>
         public override string Translate(PathSelectItem item)
         {
-            return string.Empty;
+            NodeToStringBuilder nodeToStringBuilder = new NodeToStringBuilder();
+            string res = string.Empty;
+            if (item.FilterOption != null)
+            {
+                res += "$filter=" + nodeToStringBuilder.TranslateFilterClause(item.FilterOption);
+            }
+
+            if (item.OrderByOption != null)

```

Comment:
@KenitoInc Would switching `if` with `else if` in this and subsequent blocks help reduce the number of comparisons done?